Automattic/woocommerce-payments
# Add missing VAT information prompt when downloading invoice

Fixes #3933

#### Changes proposed in this Pull Request

This PR adds a new `hasSubmittedVatData` flag to the `wcpaySettings.accountStatus` object, based on the value we get from the basic account info call. This value is then used when merchants click on the download button for a VAT invoice.

If the data has already been submitted, the document should be downloaded right away. If the data hasn't been submitted, a modal should pop up with the form to submit the VAT details.

We're only handling the modal as part of this PR. The actual form and data submission will be handled in a separate PR for #3934.

Note: A few small fixes like removing unused props from the Document type, removing a few extra `break;` statements, and re-adding the document data methods to the `wordpress/data` store have also been included in this PR.

#### Testing instructions





* Create a new `vat_invoice` for your account if you don't have one by running the following command with 1803-gh-Automattic/woocommerce-payments-server
    ```
    wp wcpay add_document_for_account {Your account id} vat_invoice --date="2022-03-01" --period_from="2022-02-01 00:00:00" --period_to="2022-02-28 23:59:59"
    ```
* Apply this PR's branch and navigate to the documents list
* Click on the download button for the VAT invoice and make sure a modal pops up
    * _The modal is pretty simple for now and should just say VAT details/VAT form_
* Apply 1896-gh-Automattic/woocommerce-payments-server to your local server or sandbox, and refresh the cached account data
* Refresh the document's list page and click on the download button
    * If your VAT data hasn't been submitted, the same modal should pop up
        * Please use the instructions on the server PR to add the VAT details for your account and test the next step
    * If you have already submitted the data, the invoice should be downloaded
        * _The download will fail while 1803-gh-Automattic/woocommerce-payments-server is not merged, but trying to access the page should be enough to test this PR_ 


-------------------


**Post merge**
